src/libsyntax/feature_gate.rs

```diff
@@ -2104,6 +2108,20 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
             ast::ExprKind::Async(..) => {
                 gate_feature_post!(&self, async_await, e.span, "async blocks are unstable");
             }
+            ast::ExprKind::Await(origin, _) => {
+                match origin {
+                    ast::AwaitOrigin::FieldLike =>
+                        gate_feature_post!(&self, async_await, e.span, "async/await is unstable"),
+                    ast::AwaitOrigin::MacroLike =>
+                        gate_feature_post!(
+                            &self,
+                            await_macro,
+                            e.span,
+                            "`await!(<expr>)` macro syntax is unstable, and will soon be removed \
+                            in favor of `<expr>.await` syntax."
+                        ),
```

Comment:
Not sure if it's worth the effort, but this could be a rustfixable suggestion

Reply:
Leaving unresolved-- not going to worry about trying to make it correct to the point of automation right now-- seems like a fine thing to work on, but I'd like to land this Pr first.

Reply:
cc #60613